backend/src/main/java/fk/prof/backend/http/BackendHttpVerticle.java

```diff
@@ -117,13 +136,55 @@ private void handlePostProfile(RoutingContext context) {
         });
   }
 
+  private void handlePostPoll(RoutingContext context) {
+    try {
+      Recorder.PollReq pollReq = ProtoUtil.buildProtoFromBuffer(Recorder.PollReq.parser(), context.getBody());
+      Recorder.ProcessGroup processGroup = RecorderProtoUtil.mapRecorderInfoToProcessGroup(pollReq.getRecorderInfo());
+      ProcessGroupContextForPolling processGroupContextForPolling = this.processGroupDiscoveryContext.getProcessGroupContextForPolling(processGroup);
+      if(processGroupContextForPolling == null) {
+        throw new IllegalArgumentException("Process group " + RecorderProtoUtil.processGroupCompactRepr(processGroup) + " not associated with the backend");
+      }
+
+      Recorder.WorkAssignment nextWorkAssignment = processGroupContextForPolling.receivePoll(pollReq);
```

Comment:
Fixed in commit https://github.com/Flipkart/fk-prof/commit/da0053a6339586f1e0e9d9e16714d9009d2fba17